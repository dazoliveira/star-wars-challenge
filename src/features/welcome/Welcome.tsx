import { Box, Button, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    flexFlow: 'row nowrap',
    justifyContent: 'flex-start',
    alignItems: 'center',
    minHeight: '100vh'
  }
}))

export default function Welcome() {

  const classes = useStyles()

  return (
    <div className={classes.root}>
       <Box flex justifyContent='center' mt={27}>
        <Typography variant='h5'>
          Welcome to <span>IClinic</span>
        </Typography>
        <Typography>
          frontend challenge
        </Typography>
       </Box>
        <Box mt={20}>
          <Button fullWidth variant="contained" color="primary">
            Start
          </Button>
        </Box>
    </div>
  );
}
