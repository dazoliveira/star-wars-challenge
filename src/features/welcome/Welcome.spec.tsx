import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from "utils/tests/helpers";
import Welcome from './Welcome';


let location: string = ''
let welcomePage: string = ''

describe('<Welcome />', () => {

    it('should render Welcome page with start button', () => {

        const { container } = renderWithTheme(<Welcome />)

        const startButton = screen.queryByText(/S T A R T/i)

        expect(startButton).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should start button click render request result page',async () => {
        const { getByRole } = renderWithTheme(<Welcome />)

        const startButton = getByRole('button')

       await userEvent.click(startButton)

       await setTimeout(() => {
        location = window.location.pathname
        welcomePage = '/jedi-vs-darkside'
       }, 5000)

        expect(location).toEqual(welcomePage)
    })
});