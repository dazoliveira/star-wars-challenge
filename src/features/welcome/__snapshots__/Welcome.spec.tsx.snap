// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> should render Welcome page with start button 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    data-testid="brand-title"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Welcome to
         
        <span
          class="MuiTypography-root MuiTypography-h1"
        >
          iClinic
        </span>
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorPrimary MuiTypography-alignCenter"
      >
        frontend challenge
      </p>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-buttonBox-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-startButton-3 MuiButton-containedPrimary MuiButton-fullWidth"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        S T A R T
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
